src/src/PluginStructs/P104_data_struct.h

```diff
@@ -0,0 +1,440 @@
+// - include the .h file guarded by #ifdef P104_USE_MY_FANCY_FONT
+// - extend in P104_data_struct::webform_load the fontTypes, fontOptions arrays like the P104_USE_NUMERIC_DOUBLEHEIGHT_FONT example
+//   the description should include the font ID for documentation purposes
+//   don't forget to guard the extra code with #ifdef P104_USE_MY_FANCY_FONT
+// - extend in P104_data_struct::handlePluginWrite the list of supported font id's for the "font" command
+// - extend in P104_data_struct::configureZones the switch/case statement to conditionaly support the new font
+// - update documentation
+
+// This is the default font id
+# define P104_DEFAULT_FONT_ID     0
+
+// These fonts are copied from the MD_Parola examples
+# ifdef P104_USE_NUMERIC_DOUBLEHEIGHT_FONT
+
+#  define P104_DOUBLE_HEIGHT_FONT_ID 1
+
+#  include "../Static/Fonts/P104_font_numeric7SegDouble.h"
+# endif // ifdef P104_USE_NUMERIC_DOUBLEHEIGHT_FONT
+
+# ifdef P104_USE_FULL_DOUBLEHEIGHT_FONT
+
+#  define P104_FULL_DOUBLEHEIGHT_FONT_ID 2
+
+#  include "../Static/Fonts/P104_font_BigFont.h"
+# endif // ifdef P104_USE_FULL_DOUBLEHEIGHT_FONT
+
+# ifdef P104_USE_VERTICAL_FONT
+
+#  define P104_VERTICAL_FONT_ID 3
+
+#  include "../Static/Fonts/P104_font_vertical.h"
+# endif // ifdef P104_USE_VERTICAL_FONT
+
+# ifdef P104_USE_EXT_ASCII_FONT
+
+#  define P104_EXT_ASCII_FONT_ID 4
+
+#  include "../Static/Fonts/P104_font_ExtASCII.h"
+# endif // ifdef P104_USE_EXT_ASCII_FONT
+
+# ifdef P104_USE_ARABIC_FONT
+
+#  define P104_ARABIC_FONT_ID 5
+
+#  include "../Static/Fonts/P104_font_arabic.h"
+# endif // ifdef P104_USE_ARABIC_FONT
+
+# ifdef P104_USE_GREEK_FONT
+
+#  define P104_GREEK_FONT_ID 6
+
+#  include "../Static/Fonts/P104_font_greek.h"
+# endif // ifdef P104_USE_GREEK_FONT
+
+# ifdef P104_USE_KATAKANA_FONT
+
+#  define P104_KATAKANA_FONT_ID 7
+
+#  include "../Static/Fonts/P104_font_katakana.h"
+# endif // ifdef P104_USE_KATAKANA_FONT
+
+struct P104_zone_struct {
+  P104_zone_struct(uint8_t _zone) : zone(_zone) {
+    size          = 0u;
+    text          = F("\"\"");
+    alignment     = 0u;
+    animationIn   = 1u; // Doesn't allow 'None'
+    speed         = 0u;
+    animationOut  = 0u;
+    pause         = 0u;
+    font          = 0u;
+    content       = 0u;
+    layout        = 0u;
+    specialEffect = 0u;
+    offset        = 0u;
+    brightness    = -1;
+    repeatDelay   = -1;
+  }
+
+  String   text;
+  int32_t  repeatDelay;
+  uint32_t _repeatTimer;
+  uint16_t speed, pause;
+  uint8_t  zone;
+  uint8_t  size;
+  uint8_t  alignment;
+  uint8_t  animationIn, animationOut;
+  uint8_t  font;
+  uint8_t  content;
+  uint8_t  layout;
+  uint8_t  specialEffect;
+  uint8_t  offset;
+  int8_t   brightness;
+  int8_t   _lastChecked = -1;
+  # ifdef P104_USE_BAR_GRAPH
+  uint16_t _lower, _upper; // lower and upper pixel numbers
+  uint8_t  _startModule;   // starting module, end module is _startModule + size - 1
+  # endif // ifdef P104_USE_BAR_GRAPH
+};
```

Comment:
These members are only initialized if the constructor with the zone is used.
No idea if the default constructor is used, but you can prevent it being created like this in the .h file:

```c++
P104_zone_struct() = delete;
```